modules/standard/BigInteger.chpl

```diff
@@ -3942,6 +3942,79 @@ module BigInteger {
 
 
   // Miscellaneous Functions
+  /* Test whether a :record:`bigint` will fit into
+    one of the standard integer types
+
+    :arg t: The Integral type to check agianst.
+    :type t: `integral`
+  */
+  proc bigint.fitsInto(type t: integral): bool {
+    if _local {
+    return
+      if t == c_long then
+        mpz_fits_slong_p(this.mpz) != 0
+      else if t == c_ulong then
+        mpz_fits_ulong_p(this.mpz) != 0
+      else if t == c_int then
+        mpz_fits_sint_p(this.mpz) != 0
+      else if t == c_uint then
+        mpz_fits_uint_p(this.mpz) != 0
+      else if t == c_short then
+        mpz_fits_sshort_p(this.mpz) != 0
+      else if t == c_ushort then
+        mpz_fits_ushort_p(this.mpz) != 0
+      else if t == c_schar then
+        mpz_cmp_si(this.mpz, -128) >= 0 && mpz_cmp_si(this.mpz, 127) <= 0
+      else if t == c_uchar then
+        mpz_cmp_ui(this.mpz, 0) >= 0 && mpz_cmp_ui(this.mpz, 255) <= 0
+      else
+        false;
+    } else if this.localeId == chpl_nodeID {
+      return
+        if t == c_long then
+          mpz_fits_slong_p(this.mpz) != 0
+        else if t == c_ulong then
+          mpz_fits_ulong_p(this.mpz) != 0
+        else if t == c_int then
+          mpz_fits_sint_p(this.mpz) != 0
+        else if t == c_uint then
+          mpz_fits_uint_p(this.mpz) != 0
+        else if t == c_short then
+          mpz_fits_sshort_p(this.mpz) != 0
+        else if t == c_ushort then
+          mpz_fits_ushort_p(this.mpz) != 0
+        else if t == c_schar then
+          mpz_cmp_si(this.mpz, -128) >= 0 && mpz_cmp_si(this.mpz, 127) <= 0
+        else if t == c_uchar then
+          mpz_cmp_ui(this.mpz, 0) >= 0 && mpz_cmp_ui(this.mpz, 255) <= 0
+        else
+          false;
+    } else {
+      var t_ = this;
+
+      return
+        if t == c_long then
+          mpz_fits_slong_p(t_.mpz) != 0
+        else if t == c_ulong then
+          mpz_fits_ulong_p(t_.mpz) != 0
+        else if t == c_int then
+          mpz_fits_sint_p(t_.mpz) != 0
+        else if t == c_uint then
+          mpz_fits_uint_p(t_.mpz) != 0
+        else if t == c_short then
+          mpz_fits_sshort_p(t_.mpz) != 0
+        else if t == c_ushort then
+          mpz_fits_ushort_p(t_.mpz) != 0
+        else if t == c_schar then
+          mpz_cmp_si(t_.mpz, -128) >= 0 && mpz_cmp_si(t_.mpz, 127) <= 0
+        else if t == c_uchar then
+          mpz_cmp_ui(t_.mpz, 0) >= 0 && mpz_cmp_ui(t_.mpz, 255) <= 0
+        else
+          false;
+    }
+  }
+
+  deprecated "`fits_ulong_p` is deprecated -  please use `bigint.fitsInto(uint(64))` or `bigint.fitsInto(c_ulong)`` instead"
```

Comment:
Yeah, I did not think about system dependent sizes. Thanks for pointing that out!

I initially decided to include the Chapel types because of some earlier discussion on this issue: https://github.com/chapel-lang/chapel/issues/17702#issuecomment-1158232891 . But I think you're right that for the purposes of the deprecation messages, the C types alone will suffice since the old methods were solely an interface to C type sizes.

Reply:
Yeah, we definitely want to be able to use the Chapel types as well with this routine.